crates/cheatcodes/src/inspector.rs

```diff
@@ -1400,92 +1431,105 @@ impl Cheatcodes {
         }
 
         if self.use_zk_vm {
-            if let TransactTo::Call(test_contract) = ecx.env.tx.transact_to {
-                if call.bytecode_address == test_contract {
-                    info!("using evm for calls to test contract {:?}", ecx.env);
-                    return None
+            // also skip if the target was created during a zkEVM skip
+            self.skip_zk_vm =
+                self.skip_zk_vm || self.skip_zk_vm_addresses.contains(&call.target_address);
+            if self.skip_zk_vm {
+                // reset skip_zk_vm
+                self.skip_zk_vm = false;
+                info!("running call in EVM, instead of zkEVM {:#?}", call);
```

Comment:
same  here as what I said above - early return...
Perhaps it even makes sense to extract this functionality in dedicated functions (about time, I'd add),
so we can more ergonomically return early without worrying about future additions